### PR TITLE
Feature: add metrics in metasrv

### PR DESCRIPTION
**Pull request by lichuang**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Feature: add metrics framework in metasrv. 
More metrics will be added after this pr. 

## Changelog

- New Feature

## Related Issues

part of [metrics API providing requests/sec, active watchers, outstanding(uncommitted) raft logs, number of raft logs, state machine size etc. · Issue #4938 · datafuselabs/databend](https://github.com/datafuselabs/databend/issues/4938)

### CODE REVIEW

File: metasrv/src/metrics/meta_metrics.rs [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+    has_leader: Option<Gauge>,
+}
+
+static INSTANCE: OnceCell<MetaMetrics> = OnceCell::new();
+
+impl Default for MetaMetrics {
+    fn default() -> Self {
+        Self::new()
+    }
+}
```
Comment (drmingdrmer) [inline metasrv/src/metrics/meta_metrics.rs:34]: I think this can be derived.

But it's not a big deal and should not be a blocker for this PR.

**Review by drmingdrmer — APPROVED** [review]

**Review by BohuTANG — APPROVED** [review]
+1

**Comment by BohuTANG** [comment]
@mergify update

**Comment by BohuTANG** [comment]
This patch broke the CI:
```
---- api::http::cluster_state_test::test_cluster_state stdout ----
thread 'api::http::cluster_state_test::test_cluster_state' panicked at 'metametrics is not initialized', metasrv/src/metrics/meta_metrics.rs:42:24
stack backtrace:
   0:     0x5590aa0c088d - std::backtrace_rs::backtrace::libunwind::trace::h701a19af54e96518
                               at /rustc/76d770ac21d9521db6a92a48c7b3d5b2cc535941/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
```

https://github.com/datafuselabs/databend/runs/6334802517?check_suite_focus=true#step:3:954

**Review by ariesdevil — APPROVED** [review]

**Comment by BohuTANG** [comment]
> This patch broke the CI:
> 
> ```
> ---- api::http::cluster_state_test::test_cluster_state stdout ----
> thread 'api::http::cluster_state_test::test_cluster_state' panicked at 'metametrics is not initialized', metasrv/src/metrics/meta_metrics.rs:42:24
> stack backtrace:
>    0:     0x5590aa0c088d - std::backtrace_rs::backtrace::libunwind::trace::h701a19af54e96518
>                                at /rustc/76d770ac21d9521db6a92a48c7b3d5b2cc535941/library/std/src/../../backtrace/src/backtrace/libunwind.rs:93:5
> ```
> 
> https://github.com/datafuselabs/databend/runs/6334802517?check_suite_focus=true#step:3:954

cc @lichuang

**Comment by lichuang** [comment]
> https://github.com/datafuselabs/databend/runs/6334802517?check_suite_focus=true#step:3:954

fixed.

**Comment by BohuTANG** [comment]
@mergify update

**Pull request merged**